core/src/tvu.rs

```diff
@@ -276,9 +276,10 @@ impl Tvu {
             tower_storage,
         );
 
-        let warm_quic_cache_service = if connection_cache.use_quic() {
+        let connection_cache_use_quic = connection_cache.map(|cc| cc.use_quic()).unwrap_or(false);
+        let warm_quic_cache_service = if connection_cache_use_quic {
```

Comment:
Re https://github.com/anza-xyz/agave/pull/1015#discussion_r1577249013 (cc @t-nelson), I'm actually not super keen on backporting something brittle like this (if someone fiddles with `connection_cache_use_quic` without looking at the subsequent block, 💥 ). This is what I would have proposed; seems just as small a diff as existing:
```suggestion
        let warm_quic_cache_service = connection_cache.and_then(|connection_cache| {
            if connection_cache.use_quic() {
```
(Needs the L282 unwrap removed and closing symbols added at the end, but github won't let me suggest this properly)

Reply:
I will not modify the code in bp. We should keep bp and original code as close as possible. This differences is just causing overhead in maintenance.

Reply:
> I will not modify the code in bp.

I am certainly not suggesting that. But I don't feel right approving a backport that I think is brittle. It's certainly a comment I would have made on the original PR, had I been a reviewer there.

> can you add a comment to be considerate of the unwrap below, instead?

For the sake of expediency, I will approve this with such a comment added. Feel free to PR to master and then cherry-pick here, if you are concerned about process.

Reply:
I will address in master first